Sockets/BackEndMonitoringSockets/src/CServiceConnectionHandler.cpp

```diff
@@ -6,77 +6,82 @@
 #include "CServiceConnectionHandler.h"
 
 CServiceConnectionHandler::CServiceConnectionHandler(CDataReceiver json_data) :
-	m_json_data(json_data), m_can_receive_data(true)
+	m_json_data(json_data)
 {
 	InitPeerStream();
 }
 
 bool CServiceConnectionHandler::HandleEvent(const int socket_fd, EEventType type)
 {
-	bool can_continue = true;
+	bool result = false;
 	switch (type) {
 	case EEventType::REQUEST_DATA:
-		can_continue = HandleRequestEvent(socket_fd);
+		result = HandleRequestEvent(socket_fd);
 		break;
-	case EEventType::CLOSE_EVENT:
-		can_continue = HandleResponseExitEvent(socket_fd);
+	case EEventType::LOST_REQUEST:
+		result = HandleWrongRequestEvent(socket_fd);
 		break;
+	default:
+		return false;
 	}
-	return can_continue;
+
+	return result;
 }
 
 bool CServiceConnectionHandler::HandleRequestEvent(const int socket_fd)
 {
-	bool can_continue = true;
-	if (m_p_peer_stream->CanReceiveData(socket_fd))
+	bool should_not_close_client = false;
+	std::string message;
+
+	if (m_p_peer_stream->CanReceiveData(socket_fd) && 
+		m_p_peer_stream->Receive(socket_fd, message))
 	{
-		std::string message = m_p_peer_stream->Receive(socket_fd);
-		if (m_can_receive_data && message == "ALL_DATA")
+		if (IsEqualStrings(message, "ALL_DATA"))
```

Comment:
Use enum, an array of strings and method "eMessageType ParseMessageType(const string & message);"
after that call this method here:
switch(ParseMessageType(message)):
case eAllData:
  "do something";
  break;
etc.

Reply:
Fixed